app/javascript/http_api/api.js

```diff
@@ -24,11 +24,19 @@ const API = {
 
 export default API;
 
+function apiOnly(url) {
+  if (!url.match(/^\/api($|\/)/)) {
```

Comment:
this still throws for `http://example.com/api/foo`

Reply:
How? :confused:

Reply:
hm, what about...

```js
url = new URL(url, document.location.href); // same object for both /api and http://localhost/api
if (!url.pathname.match(...))
```

Reply:
(or we could drop `^` and match `/api` anywhere, but....)

Reply:
I'll try my best

Reply:
@himdel why is it a problem that we throw for the URL beginning with `http(s)`? IMO we should never do that, I don't see any case where this would be necessary.

Reply:
Well, stricly speaking it would not be necessary at all ...except that's what our API returns.

So, for example..

`app/javascript/components/miq-about-modal.jsx`
```
  return API.get('/api').then(root =>
    API.get(root.server_info.region_href).then(region =>
      API.get(root.server_info.zone_href).then(zone => ({ ...root, region, zone })).then(data =>
```

`root.server_info` is..

```
...
region_href: "http://localhost:3000/api/regions/1"
server_href: "http://localhost:3000/api/servers/16"
zone_href: "http://localhost:3000/api/zones/1"
...
```

therefore, `API.get(root.server_info.region_href)` would throw unless we strip it.

But this should work...

```js
url = new URL(url, document.location.href);
if (!url.pathname.match(/^\/api($|\/)/)) {
  throw...
```

Reply:
(that, or fix our API to return relative hrefs only, I'd be :+1: on *that* :))